samples/wifi/raw_tx_packet/src/main.c

```diff
@@ -44,17 +44,19 @@ static void wifi_set_mode(void)
 
 int main(void)
 {
-	int status;
-
 	LOG_INF("Starting %s with CPU frequency: %d MHz", CONFIG_BOARD, SystemCoreClock/MHZ(1));
 	k_sleep(K_SECONDS(1));
 
 	wifi_set_mode();
```

Comment:
shouldn't the mode also change as per the Kconfig?

Reply:
We can have 2 Kconfigs : STA_MODE_ONLY and STA_TX_INJEXTION_MODE.
And if ``STA_TX_INJEXTION_MODE`` is enabled, we can call ``wifi_set_mode`` function.

Is that what you are expecting?

Reply:
IIUC, Set mode should be called always with appropriate bits set. no?